core/src/components/datetime/datetime.tsx

```diff
@@ -1290,17 +1326,19 @@ export class Datetime implements ComponentInterface {
       }
     }
 
-    this.processMinParts();
-    this.processMaxParts();
     const hourValues = (this.parsedHourValues = convertToArrayOfNumbers(this.hourValues));
     const minuteValues = (this.parsedMinuteValues = convertToArrayOfNumbers(this.minuteValues));
     const monthValues = (this.parsedMonthValues = convertToArrayOfNumbers(this.monthValues));
     const yearValues = (this.parsedYearValues = convertToArrayOfNumbers(this.yearValues));
     const dayValues = (this.parsedDayValues = convertToArrayOfNumbers(this.dayValues));
 
-    const todayParts = (this.todayParts = parseDate(getToday()));
+    const todayParts = (this.todayParts = parseDate(getToday())!);
     this.defaultParts = getClosestValidDate(todayParts, monthValues, dayValues, yearValues, hourValues, minuteValues);
-    this.processValue(this.value);
+
+    this.processMinParts();
+    this.processMaxParts();
```

Comment:
I moved these further down to avoid an error where if a `min` or `max` is provided without a year, these functions would error out. This was because `parseMinParts` and `parseMaxParts` fall back to the current year ([link](https://github.com/ionic-team/ionic-framework/blob/f3679c132dc801c1e1aa55c28f0d94927e118f53/core/src/components/datetime/utils/parse.ts#L217)), which is passed in via `todayParts` grabbed from the datetime. Previously, `todayParts` was being set _after_ these functions were called, so it was still `undefined`.

I'm unsure why this wasn't causing problems before, but it was causing the minmax test to crash.